keyboards/kbdcraft/adam0110/keymaps/tc/keymap.c

```diff
@@ -0,0 +1,134 @@
+        RGB_MOD,  RGB_HUD, RGB_SAD, RGB_VAI, RGB_SPD, KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,    KC_NO,   KC_NO,            KC_NO,
+        KC_NO,    KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,    KC_NO,   KC_NO,   KC_NO,   KC_NO,
+        KC_NO,    KC_NO,   KC_NO,                     KC_NO,                              KC_NO,    KC_NO,   KC_NO,   KC_NO,   KC_NO
+    ),
+};
+// clang-format on
+
+void eeconfig_init_user(void) {
+    rgb_matrix_enable_noeeprom();
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+#ifdef CONSOLE_ENABLE
+    xprintf("Layer State: 0b%0b (%u)\n", state, get_highest_layer(state));
+    switch (get_highest_layer(state)) {
+        case _BASE:
+            xprintf("Layer: _BASE\n");
+            break;
+        case _AAPL:
+            xprintf("Layer: _AAPL\n");
+            break;
+        case _FN1:
+            xprintf("Layer: _FN1\n");
+            break;
+        case _FN2:
+            xprintf("Layer: _FN2\n");
+            break;
+        case _FN3:
+            xprintf("Layer: _FN3\n");
+            break;
+        default:
+            xprintf("Layer: ???\n");
+            break;
+    }
+#endif // CONSOLE_ENABLE
+
+    // Indicate layer with top-left LED color.
+    int led_num = 50;
+    rgb_matrix_set_color_all(0, 0, 0); // all off
+    switch (get_highest_layer(state)) {
+        case _BASE:
+            rgb_matrix_set_color(led_num, 0x00, 0x00, 0xFF); // blue
+            break;
+        case _AAPL:
+            rgb_matrix_set_color(led_num, 0xFF, 0x00, 0x00); // red
+            break;
+        case _FN1:
+            rgb_matrix_set_color(led_num, 0x00, 0xFF, 0x00); // green
+            break;
+        case _FN2:
+            rgb_matrix_set_color(led_num, 0xFF, 0x00, 0xFF); // magenta
+            break;
+        case _FN3:
+            rgb_matrix_set_color(led_num, 0xFF, 0xFF, 0x00); // yellow
+            break;
+        default:
+            rgb_matrix_set_color(led_num, 0xFF, 0xFF, 0xFF); // white
+            break;
+    }
+
+    return state;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case RGB_TOG:
+            if (record->event.pressed) {
+                // When toggling RGB on/off, also switch back to the empty_effect, which allows custom RGB code to take effect.
+                rgb_matrix_toggle();
+                rgb_matrix_mode_noeeprom(RGB_MATRIX_CUSTOM_empty_effect);
+            }
+            return false;
+            break;
+    }
+    return true;
+}
+
+void keyboard_post_init_user(void) {
+    debug_enable   = true;
+    debug_matrix   = false;
+    debug_keyboard = false;
+    debug_mouse    = false;
+
+    // Set to empty_effect on boot.
+    rgb_matrix_mode_noeeprom(RGB_MATRIX_CUSTOM_empty_effect);
+
+    // Set to base layer on boot. Explicitly call this to set LED indicator state.
+    layer_clear();
+    layer_on(_BASE);
+}
+
+#undef NO_DEBUG
+#undef NO_PRINT
```

Comment:
User keymaps are no longer accepted into the repo. Unless there was previously a `via` keymap for this board, then only `default` will remain.